mailet/api/src/main/java/org/apache/mailet/AttributeName.java

```diff
@@ -0,0 +1,58 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.mailet;
+
+import java.util.Objects;
+
+/** 
+ * Strong typing for attribute name, which represent the name of an attribute store in a mail.
+ * 
+ * @since Mailet API v3.2
+ */
+public class AttributeName {
+    private final String name;
+
+    public static AttributeName of(String name) {
+        return new AttributeName(name);
```

Comment:
Are all attribute names valid? `""` or `null`?

Can you add edge cases for this?